src/arena.rs

```diff
@@ -88,12 +101,32 @@ struct ClientMe {
     pause_delay: Option<u32>,
 }
 
+impl ClientMe {
+    pub fn new(rank: Option<&Rank>, game_id: Option<&GameId>) -> ClientMe {
+        ClientMe {
+            rank: rank.cloned(),
+            withdraw: false,
+            game_id: game_id.cloned(),
+            pause_delay: None,
+        }
```

Comment:
that function will eventually have 4 arguments, one for each field. Then the only difference with building `ClientMe` is that the arguments will be unnamed from the caller? not sure that's a benefit, altho the strong types make it safer